src/platform/packages/shared/kbn-esql-ast/src/definitions/utils/autocomplete/helpers.ts

```diff
@@ -749,3 +750,62 @@ export function getValidSignaturesAndTypesToSuggestNext(
     currentArg,
   };
 }
+
+/**
+ * This function provides suggestions for named parameters within a command.
+ * It suggests parameter names and values based on the current text input and available parameters.
+ *
+ * Provide a record of available parameters, where each key is a parameter name and the value is an
+ * array of suggestions for that parameter.
+ *
+ * Examples:
+ *  | COMPLETION "prompt" WITH {                    ---> suggests parameters names
+ *  | COMPLETION "prompt" WITH { "param1": "        ---> suggests parameter values
+ *  | COMPLETION "prompt" WITH { "param1": "value", ---> suggests parameter names
+ *
+ * This helper does not suggest enclosing brackets.
+ *
+ * @param innerText
+ * @param availableParams
+ */
+export function getCommandNamedParametersSuggestions(
+  innerText: string,
+  availableParams: Record<string, ISuggestionItem[]>
+): ISuggestionItem[] {
+  // Suggest a parameter entry after { or after a comma
+  if (/{\s*$/i.test(innerText) || /,\s*$/.test(innerText)) {
```

Comment:
I wonder if we can use the parser instead of regexes in this function to make it more performant